kasa/tapo/tapobulb.py

```diff
@@ -52,6 +52,9 @@ def valid_temperature_range(self) -> ColorTempRange:
 
         :return: White temperature range in Kelvin (minimum, maximum)
         """
+        if not self.is_variable_color_temp:
+            raise SmartDeviceException("Color temperature not supported")
+
```

Comment:
I think there might be a bug in the non_variable_temp test. Is it calling `variable_temp_range` property with parenthesis and that’s why you needed to add the exception raising?  Prob good to fix the test at this point also.

Reply:
test_bulb.py
```@non_variable_temp
async def test_non_variable_temp(dev: SmartBulb):
    with pytest.raises(SmartDeviceException):
        await dev.set_color_temp(2700)

    with pytest.raises(SmartDeviceException):
        dev.valid_temperature_range()

    with pytest.raises(SmartDeviceException):
        print(dev.color_temp)
```
The same check is in SmartBulb, so I assumed it was simply missing in TapoBulb.

Reply:
Raising an exception here follows the same pattern that is used in smartbulb.py, so this is just something that I forgot to add and it looks fine for me.

Reply:
Agreed but I think  `dev.valid_temperature_range()` should be ` dev.valid_temperature_range`.

Reply:
Updated without parenthesis.